x-pack/plugins/security_solution/public/siem_migrations/rules/components/rules_table/index.tsx

```diff
@@ -40,143 +40,146 @@ export interface RulesTableComponentProps {
-      try {
-        await installMigrationRules([migrationRule.id]);
-      } finally {
-        setTableLoading(false);
+export const MigrationRulesTable: React.FC<MigrationRulesTableProps> = React.memo(
+  ({ migrationId }) => {
+    const [pageIndex, setPageIndex] = useState(0);
+    const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
+    const [searchTerm, setSearchTerm] = useState<string | undefined>();
+
+    const { data: translationStats, isLoading: isStatsLoading } =
+      useGetMigrationTranslationStats(migrationId);
+
+    const {
+      data: { ruleMigrations, total } = { ruleMigrations: [], total: 0 },
+      isLoading: isDataLoading,
+    } = useGetMigrationRules({
+      migrationId,
+      page: pageIndex,
+      perPage: pageSize,
+      searchTerm,
+    });
+
+    const [selectedRuleMigrations, setSelectedRuleMigrations] = useState<RuleMigration[]>([]);
+
+    const pagination = useMemo(() => {
+      return {
+        pageIndex,
+        pageSize,
+        totalItemCount: total,
+      };
+    }, [pageIndex, pageSize, total]);
+
+    const onTableChange = useCallback(({ page, sort }: CriteriaWithPagination<RuleMigration>) => {
+      if (page) {
+        setPageIndex(page.index);
+        setPageSize(page.size);
       }
-    },
-    [installMigrationRules]
-  );
-
-  const installTranslatedRules = useCallback(
-    async (enable?: boolean) => {
-      setTableLoading(true);
-      try {
-        await installTranslatedMigrationRules();
-      } finally {
-        setTableLoading(false);
-      }
-    },
-    [installTranslatedMigrationRules]
-  );
-
-  const ruleActionsFactory = useCallback(
-    (ruleMigration: RuleMigration, closeRulePreview: () => void) => {
-      // TODO: Add flyout action buttons
-      return null;
-    },
-    []
-  );
-
-  const { rulePreviewFlyout, openRulePreview } = useRulePreviewFlyout({
-    ruleActionsFactory,
-  });
-
-  const rulesColumns = useRulesTableColumns({
-    disableActions: isTableLoading,
-    openMigrationRulePreview: openRulePreview,
-    installMigrationRule: installSingleRule,
-  });
-
-  return (
-    <>
-      <EuiSkeletonLoading
-        isLoading={isDataLoading}
-        loadingContent={
-          <>
-            <EuiSkeletonTitle />
-            <EuiSkeletonText />
-          </>
+    }, []);
+
+    const handleOnSearch = useCallback((value: string) => {
+      setSearchTerm(value.trim());
+    }, []);
+
+    const { mutateAsync: installMigrationRules } = useInstallMigrationRules(migrationId);
+    const { mutateAsync: installTranslatedMigrationRules } =
+      useInstallTranslatedMigrationRules(migrationId);
+
+    const [isTableLoading, setTableLoading] = useState(false);
+    const installSingleRule = useCallback(
+      async (migrationRule: RuleMigration, enable?: boolean) => {
+        setTableLoading(true);
+        try {
+          await installMigrationRules([migrationRule.id]);
+        } finally {
+          setTableLoading(false);
+        }
```

Comment:
Why `try/finally`, no catch?

Reply:
added catch to be sure we cover all cases here